plugins/modules/vmware_vc_infraprofile_info.py

```diff
@@ -0,0 +1,238 @@
+
+    def vc_validate_profile_task(self):
+        infra = self.api_client.appliance.infraprofile.Configs
+        import_profile_spec = self.get_import_profile_spec()
+        validate_task = infra.validate_task(import_profile_spec)
+        if "VALID" == validate_task.get_info().result.get_field("status").value:
+            self.module.exit_json(changed=False, status=validate_task.get_info().result.get_field("status").value)
+        elif "INVALID" == validate_task.get_info().result.get_field("status").value:
+            # TO-DO: move to vmware_rest_client
+            self.module.exit_json(changed=False, status=validate_task.get_info().result.get_field("status").value)
+        else:
+            # TO-DO: move to vmware_rest_client
+            self.module.fail_json(msg='Failed to validate profile status:"%s" ' % dir(validate_task.get_info().status))
+
+    def wait_for_task(self, task, poll_interval=1):
+        while task.get_info().status == "RUNNING":
+            time.sleep(poll_interval)
+
+
+def main():
+    argument_spec = VmwareRestClient.vmware_client_argument_spec()
+    argument_spec.update(
+        encryption_key=dict(type='str', required=False),
+        description=dict(type='str', required=False),
+        decryption_key=dict(type='str', required=False),
+        api=dict(type='str', required=False, choices=['list', 'export', 'import', 'validate' ]),
+        profiles=dict(type='str', required=False),
+        config_path=dict(type='str', required=False),
+    )
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
+    result = {'failed': False, 'changed': False}
+    vmware_vc_infra_profile = VcenterProfile(module)
+    if module.params['api'] == "list":
+        if module.check_mode:
+            result.update(
+                    changed=False,
+                    desired_operation='list_vc_profile_configs',
+            )
+            module.exit_json(**result)
+        vmware_vc_infra_profile.list_vc_infrafrofile_configs()
+    if module.params['api'] == "export":
+        if module.check_mode:
+            result.update(
+                    changed=False,
+                    desired_operation='export_vc_profile_configs',
+            )
+            module.exit_json(**result)
+        vmware_vc_infra_profile.vc_export_profile_task()
+    if module.params['api'] == "import":
+        if module.check_mode:
+            result.update(
+                    changed=True,
+                    desired_operation='import_vc_profile_configs',
+            )
+            module.exit_json(**result)
+        vmware_vc_infra_profile.vc_import_profile_task()
+    if module.params['api'] == "validate":
+        if module.check_mode:
+            result.update(
+                    changed=True,
+                    desired_operation='import_vc_profile_configs',
+            )
+            module.exit_json(**result)
+        vmware_vc_infra_profile.vc_validate_profile_task()
+
```

Comment:
```suggestion


```

Reply:
@Akasurde 
should remove lines 236 and 237?

Reply:
Add two lines in-between `if __name__ == '__main__':` and last function line.